include/dxc/Support/dxcapi.use.h

```diff
@@ -178,6 +193,64 @@ void WriteOperationErrorsToConsole(IDxcOperationResult *pResult,
 void WriteOperationResultToConsole(IDxcOperationResult *pRewriteResult,
                                    bool outputWarnings);
 
+class DxcDllExtValidationSupport : public DxcDllSupport {
+  // DxcDllExtValidationSupport manages the
+  // lifetime of dxcompiler.dll, while the member, m_DxilSupport,
+  // manages the lifetime of dxil.dll
+  DxcDllSupport *m_DxilSupport = nullptr;
+
+  std::string DxilDLLPathExt = "";
+  bool ExternalDxilDLLInitializationSuccess = false;
+  // override DxcDllSupport's implementation of InitializeInternal,
+  // adding the environment variable value check for a path to a dxil.dll
+  // for external validation
+  HRESULT InitializeInternal(LPCSTR dllName, LPCSTR fnName){
+
+    // Load dxcompiler.dll
+    HRESULT result = InitializeForDll(dllName, fnName);
+    // if dxcompiler.dll fails to load, then we won't try loading dxil.dll,
+    // so set this boolean to false.
+    ExternalDxilDLLInitializationSuccess = DXC_FAILED(result) ? false : true;
+    if (!ExternalDxilDLLInitializationSuccess) {
+      return result;
+    }
+
+    // now handle external dxil.dll
+    const char *envVal = std::getenv("DXC_DXIL_DLL_PATH");
+    bool ValidateInternally = false;
```

Comment:
I find the added `bool ValidateInternally` confusing and unnecessary.

Shouldn't we just return S_OK here if there's no DXIL.dll to load?

After this point, we can always use `m_DxilSupport.IsEnabled()` to determine if we are using the external validator.